### PR TITLE
refactor!: Remove legacy node selector on RoleGroup

### PR DESCRIPTION
**Pull request by sbernauer**
## Description

Remove deprecated code from https://github.com/stackabletech/operator-rs/pull/556.
Users had enough time to migrate

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
### Steps
- [ ] https://github.com/stackabletech/documentation/pull/453
- [ ] Every operator has been upgraded to new operator-rs
- [ ] Grepped for `selector:` in every operator to update tests and docs
```

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Changed
+
+- BREAKING: Remove legacy node selector on `RoleGroup` ([#652]).
```
Comment (maltesander) [inline CHANGELOG.md:9]: Is it "really" breaking if we remove deprecated stuff? If yes can you add the crd-change/breaking label?
```suggestion
- BREAKING: Remove legacy node selector on `RoleGroup` ([#652]).
```

**Review by maltesander — COMMENTED** [review]
Thanks, LGTM! But i would bet we have it in some other operators / examples (e.g. https://github.com/stackabletech/opa-operator/blob/9c8c3bbdcf731b2f2d4f6af67bec1044c64ce56d/tests/templates/kuttl/smoke/10-install-opa.yaml.j2#L37C9-L38C23). Might be worth doing a grep on all repos?

File: CHANGELOG.md (hunk 1 above)
Comment (sbernauer) [reply CHANGELOG.md:9]: I would consider it breaking - for users as well as our operators as it's going to break stuff. I think the deprecation is the call out that there will be a breaking change in the future.
Added the labels

**Comment by sbernauer** [comment]
Regarding the grep: Totally agreed, did that on the docs repo: https://github.com/stackabletech/documentation/pull/453.
Also have added an checklist item to do it for all the operators once we bump them

File: CHANGELOG.md (hunk 1 above)
Comment (nightkr) [reply CHANGELOG.md:9]: Agreed with @sbernauer. _Especially_ since we don't currently do a great job warning users that they're using deprecated features.

**Comment by sbernauer** [comment]
Lets wait for Arch meeting

**Comment by lfrancke** [comment]
@sbernauer Can you tell me what happened with this one? It has the 23.11 label but is not actually included.

**Comment by fhennig** [comment]
Maybe @maltesander can chime in?

**Comment by maltesander** [comment]
Yeah we can remove the label since this is not in 23.11. 
Otherwise the PR looks good, whenever i saw the labelselector in tests / examples i already removed it but probably a last check before merging this (or bumping the operators to a new tag) should be done / tracked.

**Comment by sbernauer** [comment]
Yeah, PR is good to merge. I wasn't allowed to merge before because of priority reasons ;)

**Review by NickLarsenNZ — APPROVED** [review]
I approve the intention :+1:

**Pull request merged**